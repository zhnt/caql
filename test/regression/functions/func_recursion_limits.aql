// Recursion limits exploration (non-tail recursive)

// Simple non-tail recursive function
function test_depth(n, depth) {
    if n <= 0 {
        return depth
    }
    let result = test_depth(n - 1, depth + 1)
    return result  // Non-tail: we do something after the recursive call
}

print("=== Recursion Limits Exploration ===")

// Test different depths to find the exact limit
print("Testing depth 3:")
let r3 = test_depth(3, 0)
print("Result: " + string(r3))

print("Testing depth 5:")
let r5 = test_depth(5, 0)
print("Result: " + string(r5))

print("Testing depth 7:")
let r7 = test_depth(7, 0)
print("Result: " + string(r7))

print("Testing depth 8:")
let r8 = test_depth(8, 0)
print("Result: " + string(r8))

