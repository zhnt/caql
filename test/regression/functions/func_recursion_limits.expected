=== Recursion Limits Exploration ===
Testing depth 3:
Result: 3
Testing depth 5:
Result: 5
Testing depth 7:
Result: 7
Testing depth 8:
Result: 8
